src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                            expression,
+                            SyntaxFactory.Token(SyntaxKind.DotToken), generic),
+                            SyntaxFactory.ArgumentList())
+                    End If
+
+                    Return expression
+                End If
+
+                Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression)
+                Dim firstLambda = CreateLambdaExpression(anonymousFunctions.First())
+                Dim secondArgumentAnonymousFunction = anonymousFunctions.Last()
+                Dim returnedValue = DirectCast(secondArgumentAnonymousFunction.Body.Operations.First(), IReturnOperation).ReturnedValue
+                Dim secondLambda As LambdaExpressionSyntax
+                If returnedValue.Kind = OperationKind.AnonymousObjectCreation Then
+                    Dim objectCreation = DirectCast(returnedValue, IAnonymousObjectCreationOperation)
+                    Dim fieldInitializers = objectCreation.Initializers.
+                        Select(Function(initializer) SyntaxFactory.InferredFieldInitializer(
+                        SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name))))
+                    Dim anonymousObjectCreation = SyntaxFactory.AnonymousObjectCreationExpression(
+                        SyntaxFactory.ObjectMemberInitializer(SyntaxFactory.SeparatedList(Of FieldInitializerSyntax)(fieldInitializers)))
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction, anonymousObjectCreation)
+                Else
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction)
+                End If
+
+                Return CreateInvocationExpression(expression, NameOf(Enumerable.SelectMany), {firstLambda, secondLambda})
+            End Function
+
+            Private Function CreateLambdaExpression(anonymousFunction As IAnonymousFunctionOperation, Optional body As VisualBasicSyntaxNode = Nothing) As LambdaExpressionSyntax
+                If body Is Nothing Then
+                    body = MakeIdentifierNameReplacements(anonymousFunction.Body.Operations.First().Syntax)
+                End If
+                Dim parameters = anonymousFunction.Symbol.Parameters.
+                    Select(Function(p) SyntaxFactory.Parameter(SyntaxFactory.ModifiedIdentifier(BeautifyName(p.Name))))
+                Dim parameterList = SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(parameters))
+                Dim header = SyntaxFactory.LambdaHeader(
+                    SyntaxKind.FunctionLambdaHeader,
+                    attributeLists:=Nothing,
+                    modifiers:=Nothing,
+                    SyntaxFactory.Token(SyntaxKind.FunctionKeyword),
+                    parameterList,
+                    asClause:=Nothing)
+                Return SyntaxFactory.SingleLineFunctionLambdaExpression(header, body)
```

Comment:
>Return SyntaxFactory.SingleLineFunctionLambdaExpression(header, body) [](start = 16, length = 69)

We should verify assumptions and ensure that the lambda has the body that we expect.